includes/import/abstract-wc-product-importer.php

```diff
@@ -0,0 +1,941 @@
+					$product->set_image_id( $attachment_id );
+				} else {
+					$gallery[] = $attachment_id;
+				}
+
+				// Set the image alt if present.
+				if ( ! empty( $image['alt'] ) ) {
+					update_post_meta( $attachment_id, '_wp_attachment_image_alt', wc_clean( $image['alt'] ) );
+				}
+
+				// Set the image name if present.
+				if ( ! empty( $image['name'] ) ) {
+					wp_update_post( array( 'ID' => $attachment_id, 'post_title' => $image['name'] ) );
+				}
+			}
+
+			if ( ! empty( $gallery ) ) {
+				$product->set_gallery_image_ids( $gallery );
+			}
+		} else {
+			$product->set_image_id( '' );
+			$product->set_gallery_image_ids( array() );
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Save product shipping data.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param array      $data    Shipping data.
+	 * @return WC_Product
+	 */
+	protected function save_product_shipping_data( $product, $data ) {
+		// Virtual.
+		if ( isset( $data['virtual'] ) && true === $data['virtual'] ) {
+			$product->set_weight( '' );
+			$product->set_height( '' );
+			$product->set_length( '' );
+			$product->set_width( '' );
+		} else {
+			if ( isset( $data['weight'] ) ) {
+				$product->set_weight( $data['weight'] );
+			}
+
+			// Height.
+			if ( isset( $data['height'] ) ) {
+				$product->set_height( $data['height'] );
+			}
+
+			// Width.
+			if ( isset( $data['width'] ) ) {
+				$product->set_width( $data['width'] );
+			}
+
+			// Length.
+			if ( isset( $data['length'] ) ) {
+				$product->set_length( $data['length'] );
+			}
+		}
+
+		// Shipping class.
+		if ( isset( $data['shipping_class_id'] ) ) {
+			$shipping_class_term = get_term_by( 'id', wc_clean( $data['shipping_class_id'] ), 'product_shipping_class' );
+
+			if ( $shipping_class_term ) {
+				$product->set_shipping_class_id( $shipping_class_term->term_id );
+			}
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Save downloadable files.
+	 *
+	 * @param WC_Product $product    Product instance.
+	 * @param array      $downloads  Downloads data.
+	 * @param int        $deprecated Deprecated since 3.0.
+	 * @return WC_Product
+	 */
+	protected function save_downloadable_files( $product, $downloads ) {
+		$files = array();
+		foreach ( $downloads as $key => $file ) {
+			if ( empty( $file['file'] ) ) {
+				continue;
+			}
+
+			$download = new WC_Product_Download();
+			$download->set_id( $key );
+			$download->set_name( $file['name'] ? $file['name'] : wc_get_filename_from_url( $file['file'] ) );
+			$download->set_file( apply_filters( 'woocommerce_file_download_path', $file['file'], $product, $key ) );
+			$files[]  = $download;
+		}
+		$product->set_downloads( $files );
+
+		return $product;
+	}
+
+	/**
+	 * Save taxonomy terms.
+	 *
+	 * @param WC_Product $product  Product instance.
+	 * @param array      $terms    Terms data.
+	 * @param string     $taxonomy Taxonomy name.
+	 * @return WC_Product
+	 */
+	protected function save_taxonomy_terms( $product, $terms, $taxonomy = 'cat' ) {
```

Comment:
This is another case where the parser should format category and tag_ids ready for import?

@claudiulodro